hdl/ip/vhd/common/utils/time_pkg.vhd

```
@@ -146,7 +146,7 @@ package body time_pkg is
     begin
         -- Not all values work out evenly since we're dividing, so we round up the value as needed
         -- to be greater than or equal to the requested value
-        ret_num := to_unsigned(natural(ceil(real(desired * sacle_factor) / real(clk_per_ns))), ret_num'length);
+        ret_num := to_unsigned(natural(ceil(real(desired * sacle_factor) / real(clk_per_ns))), 32)(return_size-1 downto 0);
```

What happens if you do this (basically allowing the to_unsigned to scale out to 32 bits and then resizing down)?
`resize(to_unsigned(natural(ceil(real(desired * sacle_factor) / real(clk_per_ns))), 32), ret_num'length)`

---

I *think* this should then only warn with truncation if the *value* of the number is smaller than the resize, which I think is what we want.

As it is, you'll silently drop bits where the number is > than the size you are returning which still seems problematic?